proposals/cleanup.md

```diff
@@ -209,12 +206,61 @@ A couple implementations have been suggested for the annotation/label methodolog
 
+
+## Implementation Approach for Label Based Resource Removal in CleanUp Policy
+
+### Metadata Informer Design and Implementation
+
+
+The goal is to design and implement a metadata informer that syncs with the informer factory. This informer focuses on extracting labels and checking for appropriate labels. Specifically, it searches for the cleanup.kyverno.io/ttl label and processes resources associated with it. The following steps outline the process:
+
+- The informer factory is established to monitor and sync resources.
+
+- The informer identifies resources with the cleanup.kyverno.io/ttl label and adds them to the factory for further processing.
+
+- A work queue is utilized to manage resources with this label.
+
+- The business logic schedules the deletion of resources labeled with a specific schedule. First of all the time is parsed into the standard format and then the timestamp is added to the current time and a countdown timer is setted up after which the deletion of the resource takes place via the client.
+
+- Upon labeling, the controller is notified and verifies the label's valid format for parsing.
+
+- The delay between the current time and the scheduled deletion time is calculated.
+
+- Relevant information for deletion is added to the work queue, timed to be visible when the calculated delay elapses.
+
+- Workers retrieve items from the queue and execute actual deletions.
+
+### Key Points
+
+- **Single Label**: Due to the limitation of only AND-based operator, a single label (cleanup.kyverno.io/ttl) is used to identify resources for deletion.
+
+- **Date Format**: The label's date format is adjusted from 2022-08-04T00:30:00Z to 2022-08-04T003000Z to comply with validation webhooks.
+
+- **Informer Factory**: Custom informers ensure graceful controller shutdown and prevent resource leaks.
+
+- **Discovery API**: A discovery API identifies resources supporting necessary verbs (get, list, delete) and skips non-compliant resources.
+
+- **Permissions**: The auth package validates service account permissions for watch, list, and delete operations.
+
+- **Controller Management**: Logic tracks resource creation, deletion, and changes in RBAC permissions, managing individual controllers.
+
+- **Graceful Shutdown**: Cancellable context is utilized for controllers and workers to ensure graceful shutdown.
+
+- **Unit Tests**: Tests cover time calculation and parsing to ensure accuracy.
+
+- **Kuttl Tests**: Comprehensive Kuttl tests cover various scenarios and controller behavior.
+
+- **Validation-Webhook**: A new validation webhook ensures that label values conform to the recommended format.
```

Comment:
Add a note to explain that it doesn't block resource creation but returns a warning.